yt_dlp/extractor/odkmedia.py

```diff
@@ -0,0 +1,177 @@
+            seoSynopsisZhHant
+            seoKeyword
+            seoKeywordEn
+            seoKeywordKo
+            seoKeywordZhHans
+            seoKeywordZhHant
+            seoTitle
+            seoTitleEn
+            seoTitleKo
+            seoTitleZhHans
+            seoTitleZhHant
+            videoCcLanguages
+            videoDuration
+            releaseDate
+            people {
+                ...PeopleItem
+                __typename
+            }
+            images {
+                thumbnail
+                __typename
+            }
+            ageRating {
+                isBlock
+                __typename
+            }
+            __typename
+            }
+
+            fragment NextEpisodeDetail on Episode {
+                program {
+                    slug
+                    __typename
+                }
+                link
+                kind
+                part
+                episodeNumber
+                images {
+                    thumbnail
+                    __typename
+                }
+                title
+                titleEn
+                titleKo
+                titleZhHans
+                titleZhHant
+                __typename
+            }
+
+            fragment PeopleItem on People {
+                role
+                slug
+                nameEn
+                nameZhHans
+                nameZhHant
+                __typename
+            }
+            """
+
+    def _get_video_info(self, episode_number, program_slug, display_id):
+        video_info = self._download_json(
+            'https://odc-graphql.odkmedia.io/graphql', display_id,
+            headers={'Content-type': 'application/json'},
+            data=json.dumps({
+                "operationName": "Episode",
+                "variables": {
+                    "programSlug": program_slug,
+                    "episodeNumber": int(episode_number),
+                    "kind": "series",
+                    "part": None
+                },
+                "query": self._QUERY}).encode())['data']['episode']
+        return video_info
+
+    def _real_extract(self, url):
+        program_slug, display_id, ep_number = self._match_valid_url(url).group(
+            'series_name', 'id', 'episode_num')
+        webpage = self._download_webpage(url, display_id)
+        video_info = self._get_video_info(ep_number, program_slug, display_id)
+        try:
+            source_json = self._download_json(
+                f'https://odkmedia.io/odc/api/v2/playback/{video_info["id"]}/', display_id,
+                headers={'Authorization': '', 'service-name': 'odc'})
+        except ExtractorError as e:
+            error_data = self._parse_json(e.cause.read(), display_id)
+            raise GeoRestrictedError(error_data.get('detail'))
+
+        formats, subtitles = [], {}
+        for source in traverse_obj(source_json, ('sources', ...)):
+            if source.get('type') == 'hls':
```

Comment:
else?

Reply:
Do we not want non-hls formats? What other formats are there? If we can't deal with them, maybe a warning could be issued?

Reply:
i don't have any other link to test as most of their video are geo-blocked. The link on the test only give hls videos, so i am not sure is there any non hls videos is available ?

Reply:
Applied the suggestion in 9bfd2d10c16756dc76990440d571c52f7b1608d9